docs/src/docs/asciidoc/gradle-plugin.adoc

```diff
@@ -173,58 +185,65 @@ The following configuration options are available for building images:
-  mainClass.set("org.test.Main") // The main class to use, defaults to the application.mainClass
-  debug.set(true) // Determines if debug info should be generated, defaults to false
-  verbose.set(true) // Add verbose output, defaults to false
-  fallback.set(true) // Sets the fallback mode of native-image, defaults to false
-  sharedLibrary.set(false) // Determines if image is a shared library, defaults to false if `java-library` plugin isn't included
-
-  systemProperties.putAll(mapOf(name1 to "value1", name2 to "value2")) // Sets the system properties to use for the native image builder
-  configurationFileDirectories.from(file("src/my-config")) // Adds a native image configuration file directory, containing files like reflection configuration
-
-  // Advanced options
-  buildArgs.add("-H:Extra") // Passes '-H:Extra' to the native image builder options. This can be used to pass parameters which are not directly supported by this extension
-  jvmArgs.add("flag") // Passes 'flag' directly to the JVM running the native image builder
-
-  // Runtime options
-  runtimeArgs.add("--help") // Passes '--help' to built image, during "nativeRun" task
-
-  // Development options
-  agent.set(true) // Enables the reflection agent. Can be also set on command line using '-Pagent'
-
-
-  useFatJar.set(true) // Instead of passing each jar individually, builds a fat jar
+graalvmNative {
+  binaries {
+    main {
+      // Main options
+      imageName.set("application") // The name of the native image, defaults to the project name
+      mainClass.set("org.test.Main") // The main class to use, defaults to the application.mainClass
+      debug.set(true) // Determines if debug info should be generated, defaults to false
+      verbose.set(true) // Add verbose output, defaults to false
+      fallback.set(true) // Sets the fallback mode of native-image, defaults to false
+      sharedLibrary.set(false) // Determines if image is a shared library, defaults to false if `java-library` plugin isn't included
+
+      systemProperties.putAll(mapOf(name1 to "value1", name2 to "value2")) // Sets the system properties to use for the native image builder
+      configurationFileDirectories.from(file("src/my-config")) // Adds a native image configuration file directory, containing files like reflection configuration
+
+      // Advanced options
+      buildArgs.add("-H:Extra") // Passes '-H:Extra' to the native image builder options. This can be used to pass parameters which are not directly supported by this extension
+      jvmArgs.add("flag") // Passes 'flag' directly to the JVM running the native image builder
+
+      // Runtime options
+      runtimeArgs.add("--help") // Passes '--help' to built image, during "nativeRun" task
+
+      // Development options
+      agent.set(true) // Enables the reflection agent. Can be also set on command line using '-Pagent'
+
+      useFatJar.set(true) // Instead of passing each jar individually, builds a fat jar
+    }
+  }
 }
```

Comment:
Do we need an example somewhere on how to configure an additional native image build other than `main`?

Reply:
I think this should be done in a separate PR with an API allowing to register a new image from a different source set.